### DISCUSSION TITLE
Does GoCron support the following capabilities ?

### DISCUSSION THREAD

**Discussion by bhushanibm** [go-co-op/gocron#366] [General]
Does GoCron support the following capabilites ?

1. Does it persist the Cron Expression or Scheduling Information in any database or is it in memory ?
2. Lets say if we register some job whose time is elapsed, does it trigger those jobs whose time have elapsed or it will only trigger for next iteration ?
3. Does it support job which are scheduled in timezone which has daylight scheme in to effect ?
4. Lets says if Gocron is used in Kubernetes and is packaged along with pod. If scaling of pod happens, does job gets triggered multiple times ?

**Comment by JohnRoesler** [comment]
1. persisted in memory. There is a ticket to add a way to serialize the scheduler to allow saving it to a DB, but nothing implemented today. #140 
2. you're talking about registering a job say to run at 2pm daily, but it's added at 3pm? by default it should run the job immediately and then wait for the next occurrence. You can toggle this behavior when registering a [job](https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForSchedule) or for all jobs on the [scheduler](https://pkg.go.dev/github.com/go-co-op/gocron#Scheduler.WaitForScheduleAll)
3. Yes. It uses the native golang timezone loading which relies on the host system. So, assuming the host system / docker container / etc. syncs it's time properly the jobs will run at the right times. I will note that I have seen a couple issues before where the docker container didn't properly handle timezone changes for daylight time. #69 #318 
4. If you scale it up yes, it will duplicate the jobs. It's not designed as a distributed scheduler. There is a ticket for a distributed system but it hasn't been implemented. #181

Hope that help!

**Comment by bhushanibm** [comment]
Thanks for the reply. Couple of more question/clarification which will help us to understand the library capability better.

1.  If this library is used in Kubernetes Env, it would get packaged as part of Docker pod. There might be case where the pod might be down might be for some patch upgrade. During those scenario when the pod comes back, will it trigger those jobs which were scheduled when pod was down ? This is more on clarification for my earlier question no 2.
2. When I say timezone support, I mean let say the kubernetes env which is running in asia pacific timezone. Does it support recurring daily jobs to run with US timezone and also support daylight for those job where in library modifies the trigger time based on its effect ?
3. Is there any limit on number of concurrent jobs which can be triggerred? I know we can set MaxConcurrentLimit. But will the library scale up to fire 5K at the same moment ?

**Comment by JohnRoesler** [comment]
1. When it comes back up it would take all the jobs and run them immediately (unless you have that setting disabled) and then it will calculate when the job should run next.
2. When you create the scheduler you set the timezone https://pkg.go.dev/github.com/go-co-op/gocron#NewScheduler and will operate in that timezone. If you need different timezones per job you can spin up as many schedulers as you need.
3. Only limitation is your compute. gocron is quite light on compute as i golang.
